all datatypes for blackhole adapter

JesBei opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE master BRANCH

- Are you adding documentation?
  - TARGET THE master BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE master BRANCH

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE master BRANCH

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->

```PHP
...could not be created. Reason: The storage adapter "Zend\Cache\Storage\Adapter\BlackHole" requires a serializer plugin; please see https://docs.zendframework.com/zend-cache/storage/plugin/#quick-start for details on how to attach the plugin to your adapter.
```


```PHP
... could not be created. Reason: The adapter 'Zend\Cache\Storage\Adapter\BlackHole' doesn't implement 'Zend\EventManager\EventsCapableInterface' and therefore can't handle plugins
```

Blackhole adapter now accepts all datatypes and doesn't need a serializer anymore. Otherwise you have to implement the EventSystem but I don't think that's necessary because blackhole doesn't save anything anyway.

JesBei committed:
all datatypes for blackhole adapter

Signed-off-by: JesBei <jesper.beisner@yahoo.de>

michalbundyra reviewed (changes requested):
@JesBei Thanks for your contribution.

In order to accept the PR we need unit test to be added to cover the changes.
Also please revert all unrelated formatting changes and check DCO status. Thanks.

fabiang commented:
Btw: the error described by @JesBei happens when using the PSR-16 decorator, which leaves the Blackhole adapter unusable together with PSR-16.

JesBei commented:
> @JesBei Thanks for your contribution.
> 
> In order to accept the PR we need unit test to be added to cover the changes.
> Also please revert all unrelated formatting changes and check DCO status. Thanks.

Added the Unit Test and reverted the formatting.

boesing commented:
@JesBei Could you please add integration tests for Blackhole aswell?
There are several integration tests in the `tests/Psr` directory, you can just copy & paste one of these and apply changes to use the Blackhole adapter.

As @fabiang states, there is an issue with the PSR decorators. Thus, with those tests, we would have initial failing tests aswell.

I will move those changes to [laminas/laminas-cache-storage-adapter-blackhole](https://github.com/laminas/laminas-cache-storage-adapter-blackhole) in case we are fixing that "issue" in a patch version. If not, these changes will definitely be part of the 2.10 milestone.

fabiang commented:
> @JesBei Could you please add integration tests for Blackhole aswell?
> There are several integration tests in the `tests/Psr` directory, you can just copy & paste one of these and apply changes to use the Blackhole adapter.

The integration tests does make much sense, since the blackhole adapter doesn't cache anything and as a result the tests that come from https://github.com/php-cache/integration-tests should fail. The only reasonable test could be that the decorators can handle the adapter and imho this could be done in the unit test directly.

boesing commented:
@JesBei I have to revert my latest statement, that this will be definetely be part of the next minor (2.10.0).
The next minor version (2.10.0) of laminas-cache will introduce satellite packages for all storage adapters. Thus, I have to close PRs in here which point to adapters.

If you still need support for this feature, please fork [laminas/laminas-cache-storage-adapter-blackhole](https://github.com/laminas/laminas-cache-storage-adapter-blackhole) and re-open a PR in the adapter project.
